main.py

```diff
@@ -854,6 +857,29 @@ def run(
                 break
 
 
+def run_async_update_with_retries(mgr, current_block, logging_path, max_retries=5):
+    failed_async_calls = 0
+
+    while failed_async_calls < max_retries:
+        try:
+            async_update_pools_from_contracts(mgr, current_block, logging_path)
+            return  # Successful execution
+        except AyncUpdateRetryException as e:
+            failed_async_calls += 1
+            remaining_pools = mgr.update_remaining_pools()
+            shuffle(remaining_pools)  # shuffle to avoid repeated immediate failure of the same pool
```

Comment:
Why is shuffling applied only starting from the first failed attempt?

Sounds like it should be applied on the original list of pools, rather than (or in addition to) being applied on the list of remaining pools upon every retry.

Reply:
The assumption is that pools will not fail - this is true ~99% of the time (ie that the first attempt here will succeed). Only if while iterating through the list, one of the pools fails, and then it retries to process any remaining non-problematic pools, does shuffling become sensible. Otherwise it is just an extra operation.